index.d.ts

```diff
@@ -662,7 +662,7 @@ declare namespace Eris {
       event: "channelRecipientAdd" | "channelRecipientRemove",
       listener: (channel: GroupChannel, user: User) => void
     ): T;
-    (event: "channelUpdate", listener: (channel: AnyChannel, oldChannel: OldChannel) => void): T;
+    (event: "channelUpdate", listener: (channel: AnyChannel, oldChannel: GenericOldChannel & ({ type: 0; topic?: string; rateLimitPerUser: number } | { type: 2; bitrate: number } | { type: 5; topic?: string; rateLimitPerUser: 0 } | { type: 4 | 6 })) => void): T;
```

Comment:
Wouldn't this be cleaner if we split it up into multiple named interfaces that all inherit from `GenericOldChannel`?

Reply:
We can do that, and add ~4-5 more interfaces, it would be cleaner, yes.